maven-plugin/src/main/java/org/openmrs/maven/plugins/Run.java

```diff
@@ -115,6 +123,34 @@ public void executeTask() throws MojoExecutionException, MojoFailureException {
 		}
 	}
 
+	private void validatePort() {
+		boolean inUse = serverHelper.isPortInUse(port);
+		do {
+			if (inUse) {
+				String message = String.format("Port %s is already in use. Can we use %s? Otherwise the run will fail.", port, port + 1);
```

Comment:
I'd prefer we first determine a free port and then suggest it to the user. It can be done by calling new ServerSocket(0) and then calling getLocalPort.